src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -6106,7 +6123,20 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         }
 
         if (!s) {
-            if (intrinsic_procedures.is_intrinsic(call_name)) {
+            if (ASRUtils::IntrinsicFunctionRegistry::is_intrinsic_function(call_name)) {
```

Comment:
Just do it here instead, that way `libasr` stays the same.

"If `sin`, then something, otherwise original."